SparseIntDirectedGraph returns inconsistent edge information

While converting one of our graphs (http://law.di.unimi.it/webdata/enwiki-2013/) to a SparseIntDirectedGraph I ran into inconsistent edge information. More precisely, for some edges returned by `outgoingEdgesOf()`, `getAllEdges()` called on the edge endpoints returns the empty set, which is inconsistent, since the edge was returned by `outgoingEdgesOf()` in the first place.

There is a class reproducing this behavior: 

https://github.com/vigna/jgrapht/blob/sux4j-test/jgrapht-unimi-dsi/src/main/java/org/jgrapht/sux4j/SparseGraphTest.java

To reproduce the bug, you have to download enwiki-2013 from the link above (.graph, and .properties files). Then, generate offsets with

java it.unimi.dsi.webgraph.BVGraph -O enwiki-2013

Finally, run the class above with assertions enabled and as only argument the basename of the graph (`enwiki-2013`, if it is in the current directory):
```
java -ea org.jgrapht.sux4j.SparseGraphTest enwiki-2013
Exception in thread "main" java.lang.AssertionError: 14395206 (469192 -> 66589)
	at org.jgrapht.sux4j.SparseGraphTest.main(SparseGraphTest.java:128)
```

The test simply enumerates the edges outgoing of node 469192 and for each edge calls `getAllEdges()` between the edge endpoints.

--- a/jgrapht-opt/src/main/java/org/jgrapht/opt/graph/sparse/SparseIntDirectedGraph.java
+++ b/jgrapht-opt/src/main/java/org/jgrapht/opt/graph/sparse/SparseIntDirectedGraph.java
@@ -280,7 +280,7 @@ public class SparseIntDirectedGraph
         Iterator<Integer> it = outIncidenceMatrix.nonZerosPositionIterator(sourceVertex);
         while (it.hasNext()) {
             int eId = it.next();
-            if (getEdgeTarget(eId) == targetVertex) {
+            if (getEdgeTarget(eId).equals(targetVertex)) {
                 return eId;
             }
         }
@@ -308,7 +308,7 @@ public class SparseIntDirectedGraph
         while (it.hasNext()) {
             int eId = it.next();
 
-            if (getEdgeTarget(eId) == targetVertex) {
+            if (getEdgeTarget(eId).equals(targetVertex)) {
                 result.add(eId);
             }
         }

--- a/jgrapht-opt/src/main/java/org/jgrapht/opt/graph/sparse/SparseIntUndirectedGraph.java
+++ b/jgrapht-opt/src/main/java/org/jgrapht/opt/graph/sparse/SparseIntUndirectedGraph.java
@@ -266,7 +266,9 @@ public class SparseIntUndirectedGraph
             int v = getEdgeSource(eId);
             int u = getEdgeTarget(eId);
 
-            if (v == sourceVertex && u == targetVertex || v == targetVertex && u == sourceVertex) {
+            if (v == sourceVertex.intValue() && u == targetVertex.intValue()
+                || v == targetVertex.intValue() && u == sourceVertex.intValue())
+            {
                 return eId;
             }
         }
@@ -296,7 +298,9 @@ public class SparseIntUndirectedGraph
             int v = getEdgeSource(eId);
             int u = getEdgeTarget(eId);
 
-            if (v == sourceVertex && u == targetVertex || v == targetVertex && u == sourceVertex) {
+            if (v == sourceVertex.intValue() && u == targetVertex.intValue()
+                || v == targetVertex.intValue() && u == sourceVertex.intValue())
+            {
                 result.add(eId);
             }
         }

